examples/triangle_intersection/triangle_intersection.cpp

```diff
@@ -16,358 +16,98 @@
-
-  // Execute the queries
-  int const n = points.size();
-  Kokkos::View<int *, MemorySpace> offsets("Example::offsets", n);
-  Kokkos::View<ArborX::Point *, MemorySpace> coefficients(
-      "Example::coefficients", n);
-
-  tree.query(execution_space, points,
-             TriangleIntersectionCallback<MemorySpace>{triangles, offsets,
-                                                       coefficients});
-
-  // Check the results
-  bool success;
-  Kokkos::parallel_reduce(
-      Kokkos::RangePolicy<ExecutionSpace>(execution_space, 0, n),
-      KOKKOS_LAMBDA(int i, bool &update) {
-        constexpr float eps = 1.e-3;
-
-#if KOKKOS_VERSION >= 40200
-        using Kokkos::printf;
-#elif defined(__SYCL_DEVICE_ONLY__)
-        using sycl::ext::oneapi::experimental::printf;
-#endif
-        if (offsets(i) != i)
-        {
-          printf("Offsets are wrong for query %d.\n", i);
-          update = false;
-        }
-        auto const &c = coefficients(i);
-        auto const &t = triangles(offsets(i));
-        auto const &p_ref = points(i);
-        Point p{c[0] * t.a[0] + c[1] * t.b[0] + c[2] * t.c[0],
-                c[0] * t.a[1] + c[1] * t.b[1] + c[2] * t.c[1]};
-        if ((Kokkos::abs(p[0] - p_ref[0]) > eps) ||
-            Kokkos::abs(p[1] - p_ref[1]) > eps)
-        {
-          printf("Coefficients are wrong for query %d.\n", i);
-          update = false;
-        }
-      },
-      Kokkos::LAnd<bool, Kokkos::HostSpace>(success));
-  std::cout << "Check " << (success ? "succeeded" : "failed") << '\n';
-  return success ? EXIT_SUCCESS : EXIT_FAILURE;
+  using MemorySpace = ExecutionSpace::memory_space;
+
+  using Point = ArborX::ExperimentalHyperGeometry::Point<2>;
+  using Triangle = ArborX::ExperimentalHyperGeometry::Triangle<2>;
+
+  // Vertices:
+  // 6_____7_____8
+  // |\    |\    |
+  // |  \  |  \  |
+  // 3____\4____\5
+  // |\    |\    |
+  // |  \  |  \  |
+  // 0____\1____\2
+  std::vector<Point> v(9);
+  v[0] = {0, 0};
+  v[1] = {1, 0};
+  v[2] = {2, 0};
+  v[3] = {0, 1};
+  v[4] = {1, 1};
+  v[5] = {2, 1};
+  v[6] = {0, 2};
+  v[7] = {1, 2};
+  v[8] = {2, 2};
+
+  // Triangles:
+  // _____________
+  // |\  5 |\  7 |
+  // |4 \  |6 \  |
+  // |____\|____\|
+  // |\  1 |\  3 |
+  // |0 \  |2 \  |
+  // |____\|____\|
+  Kokkos::View<Triangle *, MemorySpace> triangles("Example::triangles", 8);
+  auto triangles_host = Kokkos::create_mirror_view(triangles);
+  triangles_host[0] = {v[0], v[1], v[3]};
+  triangles_host[1] = {v[1], v[4], v[3]};
+  triangles_host[2] = {v[1], v[2], v[4]};
+  triangles_host[3] = {v[2], v[5], v[4]};
+  triangles_host[4] = {v[3], v[4], v[6]};
+  triangles_host[5] = {v[4], v[7], v[6]};
+  triangles_host[6] = {v[4], v[5], v[7]};
+  triangles_host[7] = {v[5], v[8], v[7]};
+  Kokkos::deep_copy(triangles, triangles_host);
+
+  // Query points:
+  // _____________
+  // |\  x |\   x|
+  // |  \  |  \  |
+  // |____\|____\|
+  // |\    |\    |
+  // |  \ x|x \  |
+  // _____\|____\|
```

Comment:
You didn't want to number them?

Reply:
I tried it, but found that it was hard to distinguish from triangles. It seemed like it you need both `x` and number for clarity. So, I decided to drop the numbers.